lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
Yes, the xl.py file should not have any references to xl_r names, that's the entire purpose of this new xl_r compiler definition, to separate the _r from the non _r stanzas.